### ISSUE TITLE
Implementing the softmax activation function

### ISSUE THREAD

**Issue by Kerollmops** [c0dearm/mushin#3]
I saw that I will need [the softmax activation function](https://en.wikipedia.org/wiki/Softmax_function) for my basic word2vec neural network. The thing is the softmax activation function needs the whole list of output values and [the `Activation` trait does only know about the current output value](https://github.com/c0dearm/mushin/blob/01ca6057290d543f4e162994a08cb4b5d8aa72c3/mushin/src/activations.rs#L1-L3). I would say that I must change the trait to give the whole list of values somehow but I find it not performant enough if I continue to call it on each output value, computing the exponential sum every-time.

**Comment by c0dearm** [comment]
Hey! Sorry for the late response.

I've just released a new version of the library which basically changes it completely so probably the issue is not relevant anymore.

Adding new operations/layers should be easy peasy now.
